samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -122,60 +244,101 @@ static int lc3_enable(const struct bt_audio_codec_cfg *codec_cfg)
 
 	frames_per_sdu = bt_audio_codec_cfg_get_frame_blocks_per_sdu(codec_cfg, true);
 
-	lc3_decoder = lc3_setup_decoder(frame_duration_us, freq_hz, 0, /* No resampling */
-					&lc3_decoder_mem);
+#if defined(CONFIG_USB_DEVICE_AUDIO)
+	stream->lc3_decoder = lc3_setup_decoder(frame_duration_us, freq_hz, USB_SAMPLE_RATE,
+					&stream->lc3_decoder_mem);
+#else
+	stream->lc3_decoder = lc3_setup_decoder(frame_duration_us, freq_hz, 0,
+					&stream->lc3_decoder_mem);
+#endif /* defined(CONFIG_USB_DEVICE_AUDIO) */
 
-	if (lc3_decoder == NULL) {
+	if (stream->lc3_decoder == NULL) {
 		printk("ERROR: Failed to setup LC3 decoder - wrong parameters?\n");
 		return -1;
 	}
 
+	k_thread_start(decoder_tid);
+
 	return 0;
 }
+#endif /* defined(CONFIG_LIBLC3) */
 
-static void lc3_decode_handler(struct k_work *work)
+#if defined(CONFIG_USB_DEVICE_AUDIO)
+/* Duplicate the audio from one channel and put it in both channels */
+void mix_mono_to_stereo(void)
 {
-	int err = 0;
-	int offset = 0;
-	uint8_t *buf_data;
-	struct net_buf *ptr_net_buf;
-	int octets_per_frame;
-	struct broadcast_sink_stream *sink_stream = CONTAINER_OF(
-		k_work_delayable_from_work(work), struct broadcast_sink_stream, lc3_decode_work);
+	int16_t usb_audio_data[MAX_NUM_SAMPLES_MONO] = {0};
+	int16_t usb_audio_data_stereo[MAX_NUM_SAMPLES_STEREO] = {0};
+	uint32_t size;
+
+	size = ring_buf_get(&streams[0].audio_ring_buf, (uint8_t *)usb_audio_data,
+			    MAX_NUM_SAMPLES_STEREO);
+	if (size != MAX_NUM_SAMPLES_STEREO) {
+		memset(&((uint8_t *)usb_audio_data)[size], 0, MAX_NUM_SAMPLES_STEREO - size);
```

Comment:
With `MAX_NUM_SAMPLES_STEREO - size` you are mixing a count and a size which doesn't sound right